### PR TITLE
feat: support partially columns reading for native index data.

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Use `BlockReader` to support partially reading index data of native format.

### CODE REVIEW

**Review by ariesdevil — APPROVED** [review]
LGTM

**Comment by RinChanNOWWW** [comment]
There are some bugs need to be fixed.

File: src/query/storages/fuse/src/io/read/agg_index/agg_index_reader_native.rs [hunk 1]
```diff
@@ -12,62 +12,158 @@
...
+        let page_num = all_columns_arrays[0].len();
+        let mut blocks = Vec::with_capacity(page_num);
+
+        for i in 0..page_num {
+            let mut arrays = Vec::with_capacity(all_columns_arrays.len());
+            for array in all_columns_arrays.iter() {
+                arrays.push(array[i].clone());
+            }
+            let chunk = Chunk::new(arrays);
+            let block = DataBlock::from_arrow_chunk(&chunk, &self.reader.data_schema())?;
```
Comment (sundy-li) [inline src/query/storages/fuse/src/io/read/agg_index/agg_index_reader_native.rs:163]: There is `batch_read_array` api in strawboat.

https://github.com/sundy-li/strawboat/blob/71acd5250c664603b51767a5b89f5515cd49753a/examples/strawboat_batch_read.rs

**Comment by BohuTANG** [comment]
askbend:summary

File: src/query/storages/fuse/src/io/read/agg_index/agg_index_reader_native.rs (hunk 1 above)
Comment (RinChanNOWWW) [reply src/query/storages/fuse/src/io/read/agg_index/agg_index_reader_native.rs:163]: `batch_read_array ` needs a raw reader, but we get a native reader with page meta.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**